torch/csrc/utils/schema_info.cpp

```diff
@@ -0,0 +1,72 @@
+#include <torch/csrc/utils/schema_info.h>
+
+namespace torch {
+namespace utils {
+
+bool SchemaInfo::isMutating(int index) const {
+  TORCH_INTERNAL_ASSERT(
+      index < schema_.arguments().size() && index >= 0,
+      "Invalid index for schema.");
+
+  return schema_.arguments()[index].alias_info() != nullptr &&
+      schema_.arguments()[index].alias_info()->isWrite();
+}
+
+bool SchemaInfo::isMutating(c10::string_view name) const {
+  c10::optional<int> index = schema_.argumentIndexWithName(name);
+  TORCH_INTERNAL_ASSERT(
+      index != c10::nullopt, "Schema has no argument named ", name);
+
+  return isMutating(*index);
+}
+
+bool SchemaInfo::areAliasing(
+    const SchemaArgument& lhs,
+    const SchemaArgument& rhs) const {
+  TORCH_INTERNAL_ASSERT(
+      (lhs.type == input && lhs.index < schema_.arguments().size() &&
+       lhs.index >= 0) ||
+          (lhs.type == output && lhs.index < schema_.returns().size() &&
+           lhs.index >= 0),
+      "Invalid index for schema.");
+  TORCH_INTERNAL_ASSERT(
+      (rhs.type == input && rhs.index < schema_.arguments().size() &&
+       rhs.index >= 0) ||
+          (rhs.type == output && rhs.index < schema_.returns().size() &&
+           rhs.index >= 0),
+      "Invalid index for schema.");
+
+  const c10::Argument lhsArg = schema_.arguments()[lhs.index];
+  const c10::Argument rhsArg = schema_.arguments()[rhs.index];
+
+  if (lhsArg.alias_info() && lhsArg.alias_info()->isWildcardAfter() &&
+      rhsArg.alias_info() && rhsArg.alias_info()->isWildcardAfter()) {
+    if (lhsArg.type()->kind() == rhsArg.type()->kind()) {
```

Comment:
https://github.com/pytorch/pytorch/blob/4ee29d6033f6ff4e1b75caa265d2646648b6636c/torch/csrc/jit/ir/alias_analysis.cpp#L86-L145 is a more complete description of the rules... not sure if this is copying the same behavior?